File: 1102/main.cpp
#include <iostream>
#include <cstring>

using namespace std;

bool check(char const* s) {
	while (*s != '\0') {
		if (!memcmp(s, "one", 3))
			s += 3;
		else  if (!memcmp(s, "puton", 5))
			s += 5;
		else  if (!memcmp(s, "outputone", 9))
			s += 9;
		else  if (!memcmp(s, "outputon", 8))
			s += 8;
		else  if (!memcmp(s, "output", 6))
			s += 6;
		else  if (!memcmp(s, "out", 3))
			s += 3;
		else  if (!memcmp(s, "inputone", 8))
			s += 8;
		else  if (!memcmp(s, "inputon", 7))
			s += 7;
		else  if (!memcmp(s, "input", 5))
			s += 5;
		else  if (!memcmp(s, "in", 2))
			s += 2;
		else
			return false;
	}
	return true;
}

int main() {
	int n;
	printf("The word: [INPUT].\n");
    printf("Number of tentative N: ");
	cin >> n;
	for (int i = 0; i < n; i++) {
		string s;
		printf("=> ");
		cin >> s;
		if (check(&s[0]))
			printf("YES\n");
		else
			printf("NO\n");
	}
}

File: 1102/main.cs
using System;

class Program
{
    static bool Check(string s)
    {
        int index = 0;
        while (index < s.Length)
        {
            switch (s.Substring(index, Math.Min(9, s.Length - index)))
            {
                case "outputone":
                    index += 9;
                    break;
                case "inputone":
                    index += 8;
                    break;
                case "outputon":
                    index += 8;
                    break;
                case "inputon":
                    index += 7;
                    break;
                case "output":
                    index += 6;
                    break;
                case "input":
                    index += 5;
                    break;
                case "puton":
                    index += 5;
                    break;
                case "one":
                    index += 3;
                    break;
                case "out":
                    index += 3;
                    break;
                case "in":
                    index += 2;
                    break;
                default:
                    return false;
            }
        }
        return true;
    }

    static void Main()
    {
        Console.WriteLine("The word: [INPUT].");
        Console.Write("Number of Tentative: ");
        int n = int.Parse(Console.ReadLine());

        for (int i = 0; i < n; i++)
        {
            Console.Write("=> ");
            string s = Console.ReadLine();

            if (Check(s))
                Console.WriteLine("YES");
            else
                Console.WriteLine("NO");
        }
    }
}
